lib/batch.js

```diff
@@ -2,53 +2,80 @@
   AgendaBatch
 */
 
-var Agenda = require('agenda');
+var AgendaBatch = function(agendaModule) {
 
-var AgendaBatch = function(dbConnection) {
+  var selfObj = this;
 
-  var selfObj = {};
-
-  var init = function(dbConnect) {
-    selfObj.agendaModule = new Agenda({ db: { address: dbConnection }});
-    selfObj.agendaModule.on('complete', selfObj.cleanBatch.bind(selfObj) );
-    selfObj.batchHash = {};
-    return selfObj;
+  var spawnNewJob = function(batchKey, data, processTime, jobName) {
+    var batchData = { key: batchKey, jobData: [] };
+    batchData.jobData.push( data );
+    return agendaModule.schedule(processTime, jobName, batchData );
   }
 
-  selfObj.batch = function(batchKey, data, processTime, jobName) {
-    if ( this.batchHash[batchKey] !== undefined ) {
-      // The key / job exists
-      var job = this.batchHash[ batchKey ];
-      job.attrs.data.jobData.push( data );
-      this.agendaModule.saveJob(job);
-    } else {
-      // The key / job has not been spawned yet.
-      var batchData = { key: batchKey, jobData: [] };
-      batchData.jobData.push( data );
-      this.batchHash[ batchKey ] = this.agendaModule.schedule(processTime, jobName, batchData );
-    }
+  var updateExistingJob = function( job, data ) {
+    job.attrs.data.jobData.push( data );
+    agendaModule.saveJob(job);
   }
 
-  selfObj.define = function(jobName, options, processor) {
-    if (!processor) {
-      processor = options;
-      options = {};
+  selfObj.batch = function(batchKey, data, processTime, jobName, cb) {
+
+    if ( cb == undefined ) {
+      cb = function(err, job) {};
     }
 
-    selfObj.agendaModule.define(jobName, options, processor);
-  }
+    // REFACTOR
+    agendaModule.jobs( { 'data.key': batchKey, 'lastRunAt': { '$exists': false }  }, function(err, jobs) {
```

Comment:
is there a running flag that you could check against too?  might simplify the logic below and guarantee we don't append to a running job